src/test/ui/hygiene/impl_items-2.rs

```diff
@@ -0,0 +1,26 @@
+#![feature(decl_macro)]
+
+trait Trait {
+    fn foo() {}
+}
+
+macro trait_impl() {
+    fn foo() {}
+}
+
+// Check that we error on multiple impl items that resolve to the same trait item.
+impl Trait for i32 {
+    trait_impl!();
+    fn foo() {}
+    //~^ ERROR duplicate definitions with name `foo`: [E0201]
+}
+
+struct Type;
+
+// Check that we do not error with inherent impls.
+impl Type {
+    trait_impl!();
+    fn foo() {}
+}
+
+fn main() {}
```

Comment:
Would be good to have a test ensuring that if `main` calls `Type::foo`, it is the one from *outside* the macro. Or are there such tests already?

Reply:
There is already `src/test/ui/hygiene/impl_items.rs` which tests according to types.